### DISCUSSION TITLE
FAQ

### DISCUSSION THREAD

**Discussion by curieuxjy** [reinforcement-learning-kr/2021_RLKR_Drone_Delivery_Challenge_with_Unity#2] [Announcements]
1. 참가 신청 완료는 어떻게 확인하나요?
> 참가 신청서 작성시 바로 확인 메일이 가지 않습니다. 사전 참가 신청기간에 작성해주신 참가자 분들에게는 지난 10/19(화)일에 Github 업데이트 알림과 함께 참가 확인 메일을 보내드렸습니다.
> 
> 이번 2차 참가 신청을 작성해주신 분들과 앞서 사전 참가 신청을 해주신 모든 분들께 `10/31(일)`일에 챌린지 참가 안내 메일을 보내드릴 예정입니다.
> 
> 관련해서 메일을 받지 못하셨거나 확인이 안되신 분들은 rlkorea7@gmail.com으로 연락을 주시면 확인해드리겠습니다.
>

2. 참가 방식 [1] Python API 와 [2] ml-agents 내부 알고리즘으로 참여하는 2가지 방법 모두 제출이 가능한가요?
> 가능합니다.
> 
> 한 개의 팀에서 [1], [2] 방식 모두로 제출 가능하며,팀에서 제출한 모델들 중 제출 방식에 상관없이 최고점만 리더보드에 기록됩니다.
> 
> 성적에 따른 상금 수여(1등,2등)는 [1],[2] 제출 방식 상관 없이 모든 제출자들의 성적 순위에 따라 결정됩니다. 
> Unity 특별상의 경우에는 [2] 방식으로 제출한 제출자들끼리 따로 순위를 판단하여 상위 2팀에게 주어집니다.(Unity 특별상을 위한 리더보드는 따로 제공되지 않습니다.)
>

3. 모델 학습을 위해 제공된 환경을 어디까지 변경하여 사용할 수 있나요?
> 환경 설명 docs인 [rl_village_info.md](https://github.com/reinforcement-learning-kr/2021_RLKR_Drone_Delivery_Challenge_with_Unity/blob/master/docs/rl_village_info.md) 에 설명된 `Customization` 부분에서 `json` 파일로 조정할 수 있는 부분 외에 모든 변경은 할 수 없습니다. 
>
> 단 환경에서 제공하는 상태와 보상을 알고리즘 단에서 수정하여 사용하는 것은 가능합니다.
>

4. 수상 기준이 되는 최고 성적이란 무엇인가요?
> 수상의 기준이 되는 리더보드에 매겨지는 점수는 Test Environment 10개에 대한 (1)배달 완료수와 (2)소요시간을 기준으로 결정됩니다. 
> 
> 자세한 Evaluation에 대해서는 [explanation.md](https://github.com/reinforcement-learning-kr/2021_RLKR_Drone_Delivery_Challenge_with_Unity/blob/master/docs/explanation.md) 의 `Evaluation` 부분을 참고해주세요.
>

5. Unity 환경의 해상도는 어떻게 변경하나요?
> a. mlagents-learn으로 실행하는 경우. [run_with_ml-agents.md](https://github.com/reinforcement-learning-kr/2021_RLKR_Drone_Delivery_Challenge_with_Unity/blob/master/docs/run_with_ml-agents.md#%EC%A3%BC%EC%9A%94-command-option-table) 명령어 옵션을 확인 해주시기 바랍니다.
>
>     # 원하는 해상도의 Value 값을 width와 height에 넣어주시기 바랍니다.
>     mlagents-learn [Trainer_Config_Path] --env=[Env_Path] --run_id=[run_id] --width=[Value] --height=[Value]
>
> b. python script에서 실행하는 경우.
>
>     # Drone_gym class 생성시 width와 height를 조절해 주시기 바랍니다.
>     env = Drone_gym(time_scale=1.0, port=11000, width=84, height=84, filename='../RL_Drone/DroneDelivery.exe')
>
>     # Drone_gym.py를 사용하지 않는 경우, 아래 코드와 같이 직접 EngineConfigurationChannel()를 설정하시기 바랍니다.
>     self.engine_configuration_channel = EngineConfigurationChannel()
>     self.engine_configuration_channel.set_configuration_parameters(time_scale=time_scale, width=width, height=height)

**Comment by pnltoen** [comment]
확인하였습니다. 답변해주셔서 감사합니다!
